tests/providers/test_braket_quantum_task.py

```diff
@@ -47,6 +54,50 @@ def test_result(self):
         self.assertEqual(task.result().results[0].shots, 3)
         self.assertEqual(task.result().get_memory(), ["10", "10", "01"])
 
+    @patch(
+        "qiskit_braket_provider.providers.braket_quantum_task.AwsQuantumTaskBatch._retrieve_results"
+    )
+    def test_task_result_is_none(self, mock_retrieve_results):
+        """Tests result when result is None"""
+        mock_retrieve_results.return_value = [None, None]
+
+        task = BraketQuantumTask(
+            backend=BraketLocalBackend(name="default"),
+            task_id="AwesomeId",
+            tasks=[MOCK_LOCAL_QUANTUM_TASK, MOCK_LOCAL_QUANTUM_TASK],
+            shots=10,
+        )
+        task.result()
+
+        mock_retrieve_results.assert_called_once()
+
+    @patch("qiskit_braket_provider.providers.braket_quantum_task.AwsQuantumTask")
+    def test_queue_position(self, mock_aws_quantum_task):
+        """Tests queue position retrival"""
+        task = BraketQuantumTask(
+            backend=Mock(spec=AWSBraketBackend),
+            task_id="arn:aws:braket:::quantum-task/AwesomeId",
+            tasks=[Mock(spec=AwsQuantumTask)],
+            shots=10,
+        )
+        mock_aws_quantum_task.queue_position.return_value = QuantumTaskQueueInfo(
+            queue_type=QueueType.NORMAL, queue_position=1, message=None
+        )
+        task_queue = task.queue_position()
+
+        mock_aws_quantum_task.assert_called_once()
```

Comment:
Can you also assert that `queue_position` is called on `AwsQuantumTask`?

Reference: 
https://github.com/qiskit-community/qiskit-braket-provider/blob/b4d70a7157abc8bde3c2730827f15d4d0703415e/qiskit_braket_provider/providers/braket_quantum_task.py#L155